src/components/ui/calendar/baseCalendar.component.tsx

```diff
@@ -99,6 +100,14 @@ export abstract class BaseCalendarComponent<P, D = Date> extends React.Component
     return this.props.max || this.dateService.getYearEnd(this.dateService.today());
   }
 
+  componentDidUpdate(prevProps: Readonly<BaseCalendarProps<D> & P>, prevState: Readonly<State<D>>) {
```

Comment:
I guess this is not a correct place to handle this. I've tested it locally and now it's also called when the date is changed. So both `onSelect` and `onVisibleDateChange` is called.

I guess we should call it in `setState` callback in `onHeaderNavigationLeftPress` and `onHeaderNavigationRightPress` functions, like: 

```ts
private onHeaderNavigationLeftPress = (): void => {
  const visibleDate: D = this.createViewModeVisibleDate(-1);
  this.setState({ visibleDate }, () => {
    this.props.onVisibleDateChange && this.props.onVisibleDateChange(visibleDate);
  });
};
```

Reply:
Same for `onHeaderNavigationRightPress`. Could you please apply that change?

Reply:
But your code will not trigger `onVisibleDateChange` if date change was caused by selecting a different year or a different month from yearPicker/monthPicker.

In my current code, I can add a check if the year and month of new visibleDate and old visibleDate the same to prevent `onSelect` call.

Reply:
> your code will not trigger onVisibleDateChange if date change was caused by selecting a different year or a different month from yearPicker/monthPicker

I guess it shouldn't, because for that case you have `onSelect` being called. `onVisibleDateChange` should be called when you change the page of the picker, do I miss the conception?

Reply:
`onSelect` only called when user press a day component but if user changed month from monthPicker, `onSelect` not called(because didn't click a specific day) ,but current rendered days has changed. As my issue #1099  said, I need `onVisibleDateChange` call whenever user changed the current rendered days caused by such as rightNav,leftNav,fromMonthPicker   (hope you can understand, not good at english😅)